File: canJump.py
# -*- coding: utf-8 -*-
"""
Created on Sat Oct 10 01:03:58 2015

@author: weizhi
"""

class Solution(object):
    def canJump(self,nums):
        if not nums:
            return False
        pre = nums[0]
        
     #   result = [0]*len(nums)
      #  result[0] = nums[0]
        for index in range(1,len(nums)):
            if pre>0:
                pass
            else:
                return False
            curr = nums[index] 
            if pre-1>=curr:
                pre = pre-1
            else:
                pre = curr

        return True

File: leetcodeStudy_Java_Solution/leetcodeJavaSolution/src/leetcodeJavaSolution/canJump.java
package leetcodeJavaSolution;

public class canJump {
	public boolean canJump(int[] nums){
		if(nums==null){
			return true;
		}
		int pre = nums[0];
		//int curr = 0;
		for(int i=1;i<nums.length;i++){
			if(pre<=0){
				return false;
			}
			int curr = nums[i];
			if(nums[i]<pre-1){
				curr = pre-1;
			}
			pre = curr;

		}
		return true;
		
		}
	}


